internal/dataplane/kong_client_test.go

```diff
@@ -262,6 +276,24 @@ func (f *mockUpdateStrategyResolver) lastUpdatedContentForURL(url string) (sendc
 	return c, ok
 }
 
+func (f *mockUpdateStrategyResolver) eventuallyGetLastUpdatedContentForURL(
+	url string, waitTime, waitTick time.Duration, msgAndArgs ...any,
+) sendconfig.ContentWithHash {
+	var content sendconfig.ContentWithHash
+	if len(msgAndArgs) == 0 {
+		msgAndArgs = []any{"update was not called for URL " + url}
+	}
+	require.Eventually(f.t, func() bool {
```

Comment:
Nit: I can see `t` is already being stored in `mockUpdateStrategyResolver` before this PR but I'd suggest considering moving this outside of it to make its methods accept it as the first argument.

This way we can also add `t.Helper()` calls in those methods which would give more insight about the offending line numbers and be more idiomatic.

Reply:
Moved the `t` out of `mockUpdateStrategyResolver` and added `Helper()` to the methods.